File: client/src/app/core/models/view-models/sprint-detail-view-model.ts
import { EstimateType } from '@core/enums/estimate-type';
import { TaskViewModel } from './project-task-dto';
import { SprintViewModel } from './sprint-view-model';

export interface SprintDetailViewModel extends SprintViewModel {
  newTaskCount: number;
  activeTaskCount: number;
  doneTaskCount: number;
  totalEstimateValue?: number | null;
  estimateType?: EstimateType | null;
  tasks: TaskViewModel[];
}

File: server/Netptune.Core/ViewModels/Sprints/SprintDetailViewModel.cs
using Netptune.Core.Enums;
using Netptune.Core.ViewModels.ProjectTasks;

namespace Netptune.Core.ViewModels.Sprints;

public record SprintDetailViewModel : SprintViewModel
{
    public int NewTaskCount { get; init; }

    public int ActiveTaskCount { get; init; }

    public int DoneTaskCount { get; init; }

    public decimal? TotalEstimateValue { get; init; }

    public EstimateType? EstimateType { get; init; }

    public List<TaskViewModel> Tasks { get; init; } = new();
}
